Ezregx.Tests.Unit/Services/Foundations/Expressions/ExpressionServiceTests.Logic.cs

```diff
@@ -0,0 +1,55 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// ---------------------------------------------------------------
+
+using Ezregx.Models.Foundations.Expressions.Exceptions;
+using Ezregx.Services.Foundations.Expressions;
+using FluentAssertions;
+using HarmonyLib;
+using InternalMock.Extensions;
+using System;
+using System.Reflection;
+using Xunit;
+
+namespace Ezregx.Tests.Unit.Services.Foundations.Expressions
+{
+    public partial class ExpressionServiceTests
+    {
+        [Fact]
+        public void ShouldGetExpressionStart()
+        {
+            // given
+            string expectedExpressionStart = "^";
+
+            // when
+            string actualExpressionStart =
+                this.expressionService.RetrieveStartExpression();
+
+            // then
+            actualExpressionStart.Should().BeEquivalentTo(
+                expectedExpressionStart);
+        }
+
+        [Fact]
+        public void ShouldThrowServiceException()
+        {
+            // given
+            var exception = new Exception();
+
+            this.expressionService.Mock(
+                methodName: "GetStartExpression")
+                .Throws(exception);
+
+            // when
+            Action retrieveStartExpression = () =>
+            this.expressionService.RetrieveStartExpression();
+
+            // then
+            retrieveStartExpression.Should().Throw<ExpressionServiceException>();
```

Comment:
We also want to verify the inner exception `FailedExpressionServiceException`

Reply:
Done